### PR TITLE
Docs: Update spec as per latest on ref state.

### LINKED ISSUE #2501: Update Nessie spec asper #2312
update commit kernel : https://projectnessie.org/develop/kernel/
update Spec: https://projectnessie.org/develop/spec/

As per #2313

### PR DESCRIPTION
**Pull request by ajantha-bhat**
Fixes #2501

### CODE REVIEW

File: site/docs/develop/kernel.md [hunk 1]
```diff
@@ -27,10 +27,13 @@ current state of a table in a data lake. Whenever a table has changed via for ex
...
 carries the `Contents` object(s).
 
-`IcebergTable` contains the _current_ and _global_ pointer to Iceberg's table metadata plus the
-ID of the snapshot defined in the table metadata. Since Iceberg's table metadata manages information
-that must be consistent across all branches in Nessie, it is stored as so-called _global state_.
-The value of the snapshot-ID is stored per Nessie named reference (branch or tag).
+`IcebergTable` contains the _current_ pointer to Iceberg's table metadata plus the
+global ID's called _idGenerators_ that are defined in the table metadata. 
+Since Iceberg's table metadata manages information that must be consistent across all branches in Nessie, 
+required information is stored as so-called _global state_.
```
Comment (dimas-b) [inline site/docs/develop/kernel.md:33]: `information` -> `some information`? AFAIK, Iceberg table metadata contains components that are global (such as ID generators) but also components that are managed by branch.

File: site/docs/develop/spec.md [hunk 2]
```diff
@@ -102,17 +104,14 @@ for example an _append_ or _rewrite_ operation or more generally each Iceberg tr
...
 The Nessie `IcebergTable` object passed to Nessie in a [_Put operation_](#put-operation) therefore
 consists of
 
-1. the pointer to the Iceberg _table metadata_ and
-2. the ID of the _Iceberg snapshot_ within the Iceberg _table metadata_.
-
-The pointer to the Iceberg table is recorded as _Global State_ and the ID of the Iceberg snapshot
-is recorded within the _Put operation_ in a Nessie commit.
+1.the _idGenerators_ present in Iceberg _table metadata_. (so called _Global State_) and
+2.the pointer to the Iceberg _table metadata_  (so called _On Reference State_)
```
Comment (dimas-b) [inline site/docs/develop/spec.md:114]: add spaces after leading numbers?

File: site/docs/develop/spec.md [hunk 3]
```diff
@@ -103,16 +106,14 @@ a new Iceberg snapshot. Any Nessie commit refers to a particular Iceberg snapsho
 table, which translates to the state of an Iceberg table for a particular Nessie commit.
 
 Nessie needs to track Iceberg's _table metadata_ as so called _Global State_ within Nessie to
-ensure that schema evolution works as expected.
+ensure that table evolution and other operations like delete works as expected.
```
Comment (nastra) [inline site/docs/develop/spec.md:109]: nit: works -> work

**Review by nastra — APPROVED** [review]

File: site/docs/develop/spec.md (hunk 3 above)
Comment (ajantha-bhat) [inline site/docs/develop/spec.md:109]:
```suggestion
ensure that table evolution and other operations like delete work as expected.
```

Comment (ajantha-bhat) [reply site/docs/develop/spec.md]: done.

**Review by nastra — APPROVED** [review]

**Pull request merged**